userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -7,30 +7,44 @@
 import com.infodation.userservice.services.iservice.IUserService;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
 import org.springframework.stereotype.Service;
 
 import java.time.LocalDateTime;
+import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 
 @Service
 public class UserServiceImpl implements IUserService {
-
     private final UserRepository userRepository;
+    public UserServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Async
+    public CompletableFuture<Void> bulkEditUsers(List<UpdateUserDTO> usersDTO) {
```

Comment:
Please change the method name to be more readable
```suggestion
    public CompletableFuture<Void> bulkEditUsersAsync(List<UpdateUserDTO> usersDTO) {
```